cdm/misc/src/main/java/ucar/nc2/geotiff/GeotiffWriter.java

```diff
@@ -261,6 +271,91 @@ private void writeMetadata(boolean greyScale, double xStart, double yStart, doub
     geotiff.writeMetadata(imageNumber);
   }
 
+  /**
+   * Get a copy of the current colormap as a 1-D 3*256 element array (or null).
+   *
+   * All 256 red values first, then green, then blue.
+   *
+   * For these RGB triplets, 0 is minimum intensity, 65535 is maximum intensity (due to geotiff conventions).
+   * This function is intended for debugging and testing.
+   */
+  public int[] getColorTable() {
+    if (colorTable == null) {
+      return null;
+    } else {
+      return colorTable.clone();
+    }
+  }
+
+  /**
+   * Provide a colormap in the form of a mapping of the pixel value to the rgb triplet.
+   * Assumes an RGB of {255, 255, 255} for any values not specified.
+   *
+   * Pass null to unset the colorTable.
+   *
+   * For these RGB triplets, 0 is minimum intensity, 255 is maximum intensity. Values outside that range will
+   * be floored/ceilinged to the [0, 255] range. The color table is also assumed to be for pixel values
+   * between 0 and 255.
+   */
+  public void setColorTable(Map<Integer, Color> colorMap) {
+    setColorTable(colorMap, new Color(255, 255, 255));
+  }
+
+  /**
+   * Provide a colormap in the form of a mapping of the pixel value to the rgb triplet.
+   * Provide a default RGB triplet for any values not specified.
+   *
+   * Pass null to unset the colorTable.
+   *
+   * For these RGB triplets, 0 is minimum intensity, 255 is maximum intensity. Values outside that range will
+   * be floored/ceilinged to the [0, 255] range. The color table is also assumed to be for pixel values
+   * between 0 and 255.
+   */
+  public void setColorTable(Map<Integer, Color> colorMap, Color defaultRGB) {
+    if (colorMap == null) {
+        colorTable = null;
+        return;
+    }
+
+    // FIXME: This isn't quite right because this assumes that the data being written is
+    // unsigned bytes, but the tiff spec says that it should be sized to the width of
+    // the data type, but I would need to know the data type, which this writer doesn't know.
+    colorTable = new int[3*256];
+    for (int color = 0; color < 3; color++) {
+      for (int i = 0; i < 256; i++) {
+        // Scale it up to [0, 65535], which is needed by the ColorMap tag.
+        if (color == 0) {
+          colorTable[i] = colorMap.getOrDefault(i, defaultRGB).getRed() * 256;
+        } else if (color == 1) {
+          colorTable[256 + i] = colorMap.getOrDefault(i, defaultRGB).getGreen() * 256;
+        } else {
+          colorTable[512 + i] = colorMap.getOrDefault(i, defaultRGB).getBlue() * 256;
+        }
+      }
```

Comment:
Maybe I should unroll the color loop?

Reply:
it seems like you don't need the outer loop here?

Reply:
You do. The TIFF specification for the colortable is basically like a fortran-ordered array, so all of the reds first, then greens, then blues.

Reply:
But you're getting all three colors at once in the inner loop?

Reply:
Oh nvm, you are getting them one at a time. But you don't have to be? You could remove the outer loop and the inner if statements entirely

Reply:
True! I was so focused on filling up the array sequentially. No clue if there are performance issues either way.